pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/db/KitRequestShipping.java

```diff
@@ -857,26 +857,31 @@ public static void deactivateKitRequest(int dsmKitRequestId, @NonNull String dea
-            KitRequestShipping kitRequestShipping = new KitRequestShipping();
-            kitRequestShipping.setDsmKitRequestId(dsmKitRequestId);
-            kitRequestShipping.setDeactivationReason(deactivationReason);
-            kitRequestShipping.setDeactivatedDate(deactivatedDate);
-            kitRequestShipping.setKitTypeName(kitRequestDao.getKitTypeByKitRequestId(dsmKitRequestId));
-            try {
-                //todo: if multiple kits exist for same kit-request(dsmKitRequestId), we are updating all elements? edge case though
-                UpsertPainlessFacade.of(DBConstants.DDP_KIT_REQUEST_ALIAS, kitRequestShipping, ddpInstanceDto,
-                        ESObjectConstants.DSM_KIT_REQUEST_ID, ESObjectConstants.DSM_KIT_REQUEST_ID, dsmKitRequestId,
-                        new PutToNestedScriptBuilder()).export();
-            } catch (Exception e) {
-                logger.error(String.format("Error updating kit request shipping deactivate reason with dsm kit request id: %s in "
-                        + "ElasticSearch", dsmKitRequestId));
-                e.printStackTrace();
+            if (ddpInstanceDto.isESUpdatePossible()) {
+                //skip if no elastic index. ex: juniper studies/eating-disorder
+                KitRequestShipping kitRequestShipping = new KitRequestShipping();
+                kitRequestShipping.setDsmKitRequestId(dsmKitRequestId);
+                kitRequestShipping.setDeactivationReason(deactivationReason);
+                kitRequestShipping.setDeactivatedDate(deactivatedDate);
+                kitRequestShipping.setKitTypeName(kitRequestDao.getKitTypeByKitRequestId(dsmKitRequestId));
+                try {
+                    //todo: if multiple kits exist for same kit-request(dsmKitRequestId), we are updating all elements? edge case though
+                    UpsertPainlessFacade.of(DBConstants.DDP_KIT_REQUEST_ALIAS, kitRequestShipping, ddpInstanceDto,
+                            ESObjectConstants.DSM_KIT_REQUEST_ID, ESObjectConstants.DSM_KIT_REQUEST_ID, dsmKitRequestId,
+                            new PutToNestedScriptBuilder()).export();
+                } catch (Exception e) {
+                    logger.error(String.format("Error updating kit request shipping deactivate reason with dsm kit request id: %s in "
+                            + "ElasticSearch. Exception: %s", dsmKitRequestId, e.getMessage()));
+                    e.printStackTrace();
+                }
             }
-
         } else {
+            //todo check out when we need to request refund and probably remove else once we confirm its never invoked.
             if (easypostApiKey != null) {
                 KitRequestShipping.refundKit(dsmKitRequestId, easypostApiKey, ddpInstanceDto);
+                logger.info("Requested refund for kitRequest: {} ", dsmKitRequestId);
             }
+            logger.error("DDPInstanceDto null for deactivation of kitRequest: {}", dsmKitRequestId);
```

Comment:
This should go at the top of this clause so that it always gets logged (in case there is an error in the intervening code).